hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -158,56 +158,40 @@ case class EmitRegion(mb: EmitMethodBuilder[_], region: Value[Region]) {
 }
 
 object EmitValue {
-  def apply(missing: Option[Value[Boolean]], v: SValue): EmitValue = new EmitValue {
-    lazy val required: Boolean = missing match {
-      case None => true
-      case Some(m) => Code.constBoolValue(m).contains(false)
-    }
+  def apply(missing: Option[Value[Boolean]], v: SValue): EmitValue = new EmitValue(missing, v)
+}
 
-    override lazy val emitType: EmitType = EmitType(v.st, required)
+class EmitValue(missing: Option[Value[Boolean]], v: SValue) {
+  lazy val required: Boolean = missing match {
+    case None => true
+    case Some(m) => Code.constBoolValue(m).contains(false)
+  }
 
-    override def load: EmitCode = {
-      val ec = EmitCode(Code._empty,
-        if (required) const(false) else missing.get,
-        v)
-      assert(ec.required == required)
-      ec
-    }
+  lazy val emitType: EmitType = EmitType(v.st, required)
 
-    override def get(cb: EmitCodeBuilder): SCode = {
-      if (required) {
-        v
-      } else {
-        cb.ifx(missing.get, cb._fatal(s"Can't convert missing ${ st } to PValue"))
-        v
-      }
+  def valueTuple(): IndexedSeq[Value[_]] = {
```

Comment:
It looks like if the `missing` is `Some(const(false))` we include the bool in the value tuple. I think that's wrong -- should match on required instead of `missing` here?

Reply:
Thanks, good catch. I restructured things to make that kind of mistake harder to make.